### ISSUE TITLE
Weird behavior when a 3D array is initialized inside a 2D array

### ISSUE THREAD

**Issue by AGraber** [pawn-lang/compiler#341]
**Is this a BUG REPORT, FEATURE REQUEST or QUESTION?**:

* [x] Bug Report
* [ ] Feature Request
* [ ] Question

**What happened**:
The following code compiled with no error or warning whatsoever, and compiles to a rather strange result which gets demostrated below:
```C
static const FCNPC_IdleAnims[][] =
{
    {"PED", "FLEE_LKAROUND_01"},
    {"PED", "DAM_ARML_FRMFT"},
    {"PED", "DAM_ARMR_FRMFT"},
    {"FLOWERS", "FLOWER_HIT"},
    {"KISSING", "GF_CARARGUE_02"}
};

main()
{
    printf("%s, %s", FCNPC_IdleAnims[3][0], FCNPC_IdleAnims[3][8]);
    return 1;
}
```
output:
```
FLOWERS, FLOWER_HIT
```

**What you expected to happen**:
Not compile at all and throw an error about array dimensions

**How to reproduce it (as minimally and precisely as possible)**:
Compile the code above and run it

**Anything else we need to know?**:
Shoutout to Riddick#6416 ([Riddick94](http://forum.sa-mp.com/member.php?u=85372) at SA-MP Forums) for discovering this.

**Environment**:

* Operating System
Windows 10
* Compiler version
Tested on 3.10.4 and 3.10.7
* How are you invoking the compiler? Pawno, Sublime, vscode, sampctl or command-line?
sampctl
* If using sampctl, the version number
1.8.14

**Comment by pawnoholic** [comment]
Duplicate #137

**Comment by YashasSamaga** [comment]
@pawnoholic It isn't a duplicate. This issue should be triggering an error because you cannot 3D array initializer to a 2D array.

**Comment by YashasSamaga** [comment]
The bug can be generalized as follows:

```
static const arr[] = {"Asdsad"};
static const arr[][] = {{"Asdsad"}};
.
.
.
```

An error should be triggered for those cases. The compiler counts the braces processed to keep track of the dimensions processed. When it encounters a string literal, it does not see a brace and hence wrongly assumes that the string takes a single cell.
